osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacySwell.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            Anchor = Anchor.Centre;
+            Origin = Anchor.Centre;
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin, SkinManager skinManager)
+        {
+            Child = new Container
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Position = new Vector2(200f, 100f),
+
+                Children = new Drawable[]
+                {
+                    bodyContainer = new Container
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Alpha = 0,
+
+                        Children = new Drawable[]
+                        {
+                            spinnerCircle = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-circle"),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Scale = new Vector2(0.8f),
+                            },
+                            shrinkingRing = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-approachcircle") ?? skinManager.DefaultClassicSkin.GetTexture("spinner-approachcircle"),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Scale = Vector2.One,
+                            },
+                            remainingHitsCountdown = new LegacySpriteText(LegacyFont.Combo)
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Position = new Vector2(0f, 165f),
+                                Scale = Vector2.One,
+                            },
+                        }
+                    },
+                    clearAnimation = new Sprite
+                    {
+                        // File extension is included here because of a GetTexture limitation, see #21543
+                        Texture = skin.GetTexture("spinner-osu.png"),
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Position = new Vector2(0f, -165f),
+                        Scale = new Vector2(0.3f),
+                        Alpha = 0,
+                    },
+                }
+            };
+
+            clearSample = skin.GetSample(new SampleInfo("spinner-osu"));
+        }
+
+        public void AnimateSwellProgress(DrawableTaikoHitObject<Swell> swell, int numHits, SkinnableDrawable mainPiece)
+        {
+            remainingHitsCountdown.Text = $"{swell.HitObject.RequiredHits - numHits}";
+            spinnerCircle.RotateTo(180f * numHits, 1000, Easing.OutQuint);
```

Comment:
This rotation doesn't strictly match stable, but making it match is probably more hassle than it is worth, as stable is using a somewhat complicated system of velocity and per-frame acceleration that is shared between osu! spinners and taiko swells. I'm gonna look away until someone complains that this is a problem.

That said, this method is missing two other transformations from stable:

```csharp
                SpriteCircleTop.Transformations.RemoveAll(t => t.Type == TransformationType.Scale);
                SpriteCircleTop.Transformations.Add(new Transformation(TransformationType.Scale, Math.Min(0.94f, SpriteCircleTop.Scale + 0.02f), 0.8f, AudioEngine.Time,
                                                                   AudioEngine.Time + 400, EasingTypes.Out));

                SpriteBonusCounter.Transformations.RemoveAll(t => t.Type == TransformationType.Scale);
                SpriteBonusCounter.Transformations.Add(new Transformation(TransformationType.Scale, SpriteBonusCounter.Scale, 1.6f - 0.6f * (float)rotationRequirement / totalRotationRequirement, AudioEngine.Time,
                                                                   AudioEngine.Time + 60, EasingTypes.Out));
```

`SpriteCircleTop` is `spinnerCircle`, and `SpriteBonusCounter` is `remainingHitsCountdown`.